pytorch_lightning/trainer/evaluation_loop.py

```diff
@@ -136,6 +142,20 @@ def setup(self, model, max_batches, dataloaders):
         self.max_batches = max_batches
         self.num_dataloaders = self._get_num_dataloaders(dataloaders)
 
+        # wrap user batch samplers, so we can capture batch indices
+        dataloaders = [
+            LightningBatchSamplerWrapper.to_new_dataloader(dataloader)
+            if isinstance(dataloader, DataLoader) and self.testing and self.replace_batch_sampler_auto_id
+            else dataloader for dataloader in dataloaders]
+
+        # extract batch samplers
+        batch_samplers = [dataloader.batch_sampler for dataloader in dataloaders]
+
+        # reset batch_indices
+        self.batch_indices = None
+
+        return dataloaders, batch_samplers
```

Comment:
what is the benefit of separating the samplers from the dataloaders here?

Reply:
Moved them !